### DISCUSSION TITLE
AWS Bedrock Initialization order

### DISCUSSION THREAD

**Discussion by tomercagan** [langchain-ai/langchain-aws#220] [Ideas]
Hi,

I am using `ChatBedrock` (`BedrockBase`) / `ChatBedrockConverse` and I am wondering a bit about the initialization order (`validate_environment` and how session is created). I want to discuss specifically the usage of `credentials_profile_name`. 

Since this field is passed in in the constructor, I would expect it to be higher priority than other authentication methods. If the user provided it **explicitly**, why would it not be used?

Currently (see locations below) the classes first look at access/secret keys, and ignore the profile if they exists. In my case, I had values in the default credentials file (~/.aws/credentials) and it took it over `credentials_profile_name` that I specified (and is configured, with sso in my case, in ~/.aws/config).

I hope this makes sense and I am not missing something here. I am not that versed with all the authentication methods / deployment concerns available / used in the wild.

Here are the locations where session is initialized.  Note that in both cases, `self.credentials_profile_name` appears in a `elif` caluse:

- [BedrockBase](https://github.com/langchain-ai/langchain-aws/blob/main/libs/aws/langchain_aws/llms/bedrock.py#L624)
- [ChatBedrockConverse](https://github.com/langchain-ai/langchain-aws/blob/main/libs/aws/langchain_aws/chat_models/bedrock_converse.py#L447)

**Comment by 3coins** [comment]
@tomercagan 
We have done some updates in the recent release [v0.2.18](https://github.com/langchain-ai/langchain-aws/releases/tag/langchain-aws%3D%3D0.2.18), let us know if the latest version fixes your issue.
